politeiawww/mail/mail.go

```diff
@@ -5,104 +5,47 @@
 package mail
 
 import (
-	"crypto/tls"
-	"crypto/x509"
-	"fmt"
-	"io/ioutil"
-	"net/mail"
-	"net/url"
-
-	"github.com/dajohi/goemail"
+	"github.com/decred/politeia/politeiawww/user"
+	"github.com/google/uuid"
 )
 
-// Client provides an SMTP client for sending emails from a preset email
-// address.
-type Client struct {
-	smtp        *goemail.SMTP // SMTP server
-	mailName    string        // From name
-	mailAddress string        // From email address
-	disabled    bool          // Has email been disabled
-}
+// Mailer is an interface used to send emails to a list of recipients.
+type Mailer interface {
```

Comment:
I don't think this is quite right.  The way this is currently setup, I have to choose between a mailer that is rate limited and one that is not.  If I want to send rate limited emails to users and non-rate limited emails to the sysadmin, I'll have to have two separate mailer instances.  I should be able to send both types of emails using a single mailer instance.  Something like this:

```
// Mailer provides a set of methods for sending politeiawww emails.
type Mailer interface {
	// IsEnabled returns whether the email server is enabled.
	IsEnabled() bool

	// SendTo sends an email to a list of email addresses. These emails
	// do not have to correspond to politeiawww user emails. This
	// method is not rate limited and can be used for alternative use
	// cases like sending notifications to sysadmins.
	SendTo(subject, body string, recipients []string) error

	// SendToUsers sends an email to a list of politeiawww users,
	// identified by recipients [UUID]email map. This method rate
	// limits the number of emails that can be sent to each user.
	SendToUsers(subject, body string, recipients map[uuid.UUID]string) error
}
```

Reply:
Thanks for the quick review mate. Yep I see it now, that makes sense.

Only thing that bothers me a bit about this is that we'd have to create that recipients map from `SendToUsers` on most emails we currently send before passing it down to our mailer. This implies in a lot of code changes in a lot of places.

I suggest we add the `SendToUsers` that rate limits emails, but keep the same function signature:

`SendToUsers(subject, body string, recipients []string) error`

This way we'd only need to change `mailer.SendTo` to `mailer.SendToUsers` where it is fit, and the mailer would take care of the rest. Do you agree?

Reply:
Sure, I think that makes sense.  Lets see what that looks like.